### PR TITLE
feat(azure-handler): implement SensorData table storage (AZH-0500, AZH-0501)

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Summary

Implements **AZH-0500** and **AZH-0501** — the \SensorData\ Azure Table storage feature for enriched APP_DATA messages in the Azure handler.

## What changed

The Azure handler now extracts the \
eadings\ map (CBOR key 16, added by gateway decoder enrichment per GW-1903) from upstream \GW-0813\ APP_DATA messages and writes a \SensorData\ row for every app-data invocation.

### New types
- **\SensorDataRow\** — domain model for SensorData table rows
- **\SensorDataEntity\** — Azure Table entity with base64/hex encoding

### New functions
- **\
eadings_to_json()\** — JSON encoding with \MAX_SAFE_INTEGER\ threshold (int64 values within ±2^53−1 are JSON numbers; values exceeding that are JSON strings to preserve precision for SPA consumers using \JSON.parse\)
- **\decode_optional_readings()\** — extracts CBOR key 16 readings map from APP_DATA messages

### Modified types
- **\AppDataMessage\** — gains \
eadings: Option<BTreeMap<String, i64>>\
- **\HandlerStore\ trait** — gains \ppend_sensor_data()\
- **\RuntimeConfig\** — gains \sensor_data_table\ field (\SONDE_AZURE_HANDLER_SENSOR_DATA_TABLE\ env var)
- **\AzureTablesStore\** — gains \sensor_data_table\ client + impl
- **\decode_connector_message\** — APP_DATA now extracts readings from key 16
- **\handle_app_data\** — writes SensorData **before** ProgramRoute lookup (per design §6.1: SensorData writes are independent of routing)

### Infrastructure (AZH-0500 AC-4)
- **\storage.bicep\** — provisions \SensorData\ table resource
- **\unction-placeholder.bicep\** — sets \SONDE_AZURE_HANDLER_SENSOR_DATA_TABLE\ app setting
- **\unction-rbac.bicep\** — grants Storage Table Data Contributor to Function App identity
- **\stack.bicep\** / **\main.bicep\** — thread \sensorDataTableName\ parameter (default: \sensordata\)

## Tests added (12 new, 80 total)

| Test | Validates |
|------|-----------|
| \	_azh_0500_sensor_data_row_created_on_app_data\ | AZH-0500 |
| \	_azh_0500a_duplicate_timestamps_produce_distinct_row_keys\ | AZH-0500 |
| \	_azh_0501_decoded_readings_stored_as_json\ | AZH-0501 |
| \	_azh_0501a_missing_readings_stored_as_empty_string\ | AZH-0501 |
| \	_azh_0501b_large_int64_values_encoded_as_json_strings\ | AZH-0501 AC-5 |
| \	_azh_0501b_large_negative_int64_encoded_as_json_string\ | AZH-0501 AC-5 |
| \sensor_data_written_even_when_program_route_missing\ | Design §6.1 |
| \
eadings_to_json_*\ (5 unit tests) | Edge cases incl. \i64::MIN\ |
| \sensor_data_entity_from_row_encodes_fields_correctly\ | Entity encoding |

## Known limitations (follow-up)

- **T-AZH-0500b** (retry idempotency): requires passing upstream queue message ID through \handle_payload\ API — deferred

## Traceability

Requirements → \docs/azure-handler-requirements.md\ AZH-0500, AZH-0501, AZH-0502
Design → \docs/azure-handler-design.md\ §6.1
Validation → \docs/azure-handler-validation.md\ T-AZH-0500 through T-AZH-0501b

### CODE REVIEW

File: crates/sonde-azure-handler/src/lib.rs [hunk 1]
```diff
@@ -337,6 +358,18 @@ where
...
+        // AZH-0500: write SensorData row before ProgramRoute routing.
+        // SensorData writes are independent of routing (azure-handler-design §6.1).
+        let sensor_row = SensorDataRow {
+            row_key: next_history_row_key(app_data.timestamp_ms)?,
+            node_id: app_data.node_id.clone(),
+            timestamp_ms: app_data.timestamp_ms,
+            program_hash: app_data.program_hash.clone(),
+            raw_payload: app_data.payload.clone(),
+            decoded_readings: readings_to_json(&app_data.readings),
+        };
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-azure-handler/src/lib.rs:371]: Fixed — destructured \pp_data\ and moved fields into \SensorDataRow\ instead of cloning. ProgramRoute lookup now references \sensor_row.program_hash\.

File: crates/sonde-azure-handler/src/lib.rs [hunk 2]
```diff
@@ -814,6 +849,16 @@ impl HandlerStore for AzureTablesStore {
...
         Ok(())
     }
+
+    async fn append_sensor_data(&self, row: &SensorDataRow) -> Result<(), HandlerError> {
+        let entity = SensorDataEntity::try_from(row.clone())?;
+        self.sensor_data_table
+            .insert::<_, SensorDataEntity>(&entity)
+            .map_err(|e| HandlerError::Store(format!("prepare sensor-data insert failed: {e}")))?
+            .await
+            .map_err(|e| HandlerError::Store(format!("append sensor-data row failed: {e}")))?;
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-azure-handler/src/lib.rs:860]: The existing \ppend_actual_state\ uses the identical \TryFrom(row.clone())\ pattern (line 678). Changing only SensorData would be inconsistent with the codebase convention. Deferring to a broader refactor if needed.

File: crates/sonde-azure-handler/src/lib.rs [hunk 3]
```diff
@@ -3045,4 +3232,292 @@ mod tests {
...
+    #[test]
+    fn sensor_data_entity_conversion_round_trips() {
+        let row = SensorDataRow {
+            row_key: "000000000000fcb6:fffffffffffffffe:abcdef".to_string(),
+            node_id: "node-42".to_string(),
+            timestamp_ms: 999,
+            program_hash: vec![0x42u8; 32],
+            raw_payload: b"hello sensor".to_vec(),
+            decoded_readings: r#"{"temp":25}"#.to_string(),
+        };
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-azure-handler/src/lib.rs:3507]: Fair point — renamed to \sensor_data_entity_from_row_encodes_fields_correctly\. There is no reverse \TryFrom\ impl since SensorData rows are append-only and never read back by this handler.

File: deploy/bicep/main.bicep [hunk 4]
```diff
@@ -42,6 +42,9 @@ param programRouteTableName string = 'programroute'
 @description('Table name for program storage.')
 param programsTableName string = ''
 
+@description('Table name for sensor data rows.')
+param sensorDataTableName string = ''
+
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply deploy/bicep/main.bicep:47]: Updated the PR description to include the Bicep provisioning changes under a new Infrastructure section and removed AC-4 from the known-limitations list.

File: crates/sonde-azure-handler/src/lib.rs [hunk 5]
```diff
@@ -1405,6 +1512,40 @@ fn validate_program_hash_length(bytes: &[u8], field: &str) -> Result<(), Handler
...
+                        ))
+                    })?;
+                result.insert(name.to_string(), val);
+            }
+            Ok(Some(result))
+        }
+        Some(_) => Err(HandlerError::Decode(
+            "`readings` (key 16) must be a CBOR map or null".to_string(),
+        )),
+    }
```
[the remark this answers is not in the document]
Comment (Alan-Jowett) [reply crates/sonde-azure-handler/src/lib.rs:1546]: Added three focused tests covering each error branch: non-map value at key 16, non-text map key, and non-integer map value. 83 tests now pass.
